src/bapsiclesession/state.ts

```diff
@@ -0,0 +1,63 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { AppThunk } from "../store";
+import { Timeslot } from "../api";
+import { connectBAPSicle } from "../bapsicle";
+
+interface bapsServer {
+  hostname: String | null;
+  port: Number | null;
+  name: String | null;
+}
+
+interface sessionState {
+  currentServer: bapsServer | null; // Connection Failed, Disconnected
+  connectionState: "CONNECTED" | "CONNECTING" | "FAILED" | "DISCONNECTED";
+  currentTimeslot: Timeslot | null;
+  timeslotLoading: boolean;
+  timeslotLoadError: string | null;
+}
+
+const sessionState = createSlice({
+  name: "Session",
+  initialState: {
+    currentServer: null,
+    connectionState: "DISCONNECTED",
+    currentTimeslot: null,
+    timeslotLoading: false,
+    timeslotLoadError: null,
+  } as sessionState,
+  reducers: {
+    setCurrentServer(state, action: PayloadAction<{ server: bapsServer }>) {
+      state.currentServer = action.payload.server;
+      state.connectionState = "CONNECTING";
+    },
+    setServerState(
+      state,
+      action: PayloadAction<sessionState["connectionState"]>
+    ) {
+      state.connectionState = action.payload;
+    },
+    getState(state) {
+      return state;
+    },
+  },
+});
+
+export default sessionState.reducer;
+
+export const { setServerState } = sessionState.actions;
+
+export const getCurrentServer = (): AppThunk => async (dispatch, getState) => {
+  return getState().bapsSession.currentServer;
+};
+
+export const getServer = (): AppThunk => async (dispatch) => {
+  // TODO Server Details Configurable
```

Comment:
Reminder to file an issue for this, rather than have the TODO get lost in the codebase :p